### PR TITLE
print common error messages when builds fail

### PR DESCRIPTION
**Pull request by dsokal**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.

# Why

We want to provide users with meaningful error messages when their builds fail.

# How

I added printing the common error messages (if detected). The list of the errors (with messages and FYI URLs) is defined in `@expo/eas-build-job`.

# Test Plan

ios-failed

one-failed

two-failed

### CODE REVIEW

File: packages/eas-cli/src/build/create.ts [hunk 1]
```diff
@@ -104,7 +115,11 @@ async function waitForBuildEndAsync(
           break;
         case BuildStatus.ERRORED:
           spinner.fail('Build failed');
-          throw new Error(`Standalone build failed!`);
+          if (build.error) {
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/create.ts:118]: What about the case when building with `-p all` ?

File: packages/eas-cli/src/build/types.ts [hunk 2]
```diff
@@ -27,13 +27,19 @@ export interface Build {
...
   metadata?: Partial<Metadata>;
+  error?: BuildError;
 }
 
 interface BuildArtifacts {
   buildUrl?: string;
   logsUrl: string;
 }
 
+interface BuildError {
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/types.ts:38]: isn't it the same as ExternallError from eas-build-job ?

File: packages/eas-cli/src/build/utils/printBuildInfo.ts [hunk 3]
```diff
@@ -87,3 +99,13 @@ export function printDeprecationWarnings(deprecationInfo?: DeprecationInfo): voi
...
+
+export function printUserError(error: errors.UserError, { printNewLine = true } = {}): void {
+  if (printNewLine) {
+    Log.addNewLineIfNone();
+  }
+  Log.error(error.message);
+  if (error.fyiURL) {
+    Log.error(learnMore(error.fyiURL, { dim: false }));
+  }
+}
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/utils/printBuildInfo.ts:111]:
```suggestion
export function printUserError(error: errors.UserError): void {
  Log.error(error.message);
  if (error.fyiURL) {
    Log.error(learnMore(error.fyiURL, { dim: false }));
  }
}
```

You call this function only once with pritnNewLine true, so it's probably cleaner to add `Log.addNewLineIfNone();` there.

**Review by brentvatne — APPROVED** [review]
this will be helpful :+1:

File: packages/eas-cli/src/build/types.ts (hunk 2 above)
Comment (dsokal) [reply packages/eas-cli/src/build/types.ts:38]: Yes, it is. However, I'd prefer not to mix in that type here. The types in this file describe how the API responses look like. I think it's better to have it duplicated.

File: packages/eas-cli/src/build/create.ts (hunk 1 above)
Comment (dsokal) [reply packages/eas-cli/src/build/create.ts:118]: If you look closer at the code you notice that this `switch` is only used for building single-platform builds. I.e. it's not used for the `-p all` case.

Comment (wkozyra95) [reply packages/eas-cli/src/build/create.ts:119]: yes that is my point, we are not displaying those errors when building for multiple platforms

Comment (dsokal) [reply packages/eas-cli/src/build/create.ts:119]: Actually, we do display errors when building for multiple platforms. See also the screenshots in the PR description :P

**Review by wkozyra95 — APPROVED** [review]

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**